### PR TITLE
doc(checkout): CHECKOUT-8827 update doc to reflect new exposed "display name" field of the coupon

### PR DESCRIPTION
**Pull request by bc-shawnwang**
# [CHECKOUT-8827](https://bigcommercecloud.atlassian.net/browse/CHECKOUT-8827)

Promotion Manager now allows merchant to explicitly set display name of the promotion (for both coupon and automatic promotion). 

## What changed?

* v3 checkout management api expose display_name for each coupon object in `coupons` and `cart.coupons`
* v3 cart management api expose display_name for each coupon object in `coupons`
* for storefront cart/checkout api, adjust the wording for coupon displayName to reflect the promotion change.

### v3 cart management api
`coupons`:
Screenshot 2024-12-19 at 2 25 36 PM

### v3 checkout management api
`coupons`:
Screenshot 2024-12-19 at 2 27 27 PM

`cart.coupons`:
Screenshot 2024-12-19 at 2 28 24 PM

### storefront cart api
Screenshot 2024-12-19 at 2 32 37 PM

### storefront checkout api
`cart.coupons`:
Screenshot 2024-12-19 at 2 36 41 PM

`coupons`:
Screenshot 2024-12-19 at 2 37 47 PM

## Release notes draft

ping @bigcommerce/team-checkout @bigcommerce/dev-docs 

[CHECKOUT-8827]: https://bigcommercecloud.atlassian.net/browse/CHECKOUT-8827?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/checkouts.v3.yml [hunk 1]
```diff
@@ -8159,6 +8252,10 @@ components:
                     description: The discounted amount applied within a given context.
                     format: float
                     example: 0.9
+                  display_name:
+                    type: string
+                    description: The display name of the coupon.
+                    example: 20% Off
```
Comment (bc-shawnwang) [inline reference/checkouts.v3.yml:8258]: `cart.coupons`

File: reference/checkouts.v3.yml [hunk 2]
```diff
@@ -8863,6 +8960,10 @@ components:
           type: string
           description: Key name to identify the type of coupon.
           example: percentage_discount
+        display_name:
+          type: string
+          description: The display name of the coupon.
+          example: 20% Off
```
Comment (bc-shawnwang) [inline reference/checkouts.v3.yml:8966]: `coupons`

in `checkout.v3.yml` file, the main change is the `coupons` and `cart.coupons`, other changes are all about examples (we write exampled one by one, not use a reference to get rid of duplications, so quite a lot of examples to change)

File: reference/checkouts.sf.yml [hunk 3]
```diff
@@ -1102,7 +1102,7 @@ components:
           description: the coupon code
         displayName:
           type: string
-          description: The coupon title based on different types provided in control panel section.
+          description: The coupon name displayed on the storefront.
```
Comment (bc-shawnwang) [inline reference/checkouts.sf.yml:1105]: changed to be as same as in storefront cart api (carts.sf.yml), to be more consistent on description.

**Review by animesh1987 — APPROVED** [review]

**Comment by bc-traciporter** [comment]
Hi @bc-shawnwang should we add `display_name` to v3 Carts under line_items > physical_items > coupons > array
Screenshot 2024-12-19 at 12 34 08 PM

File: reference/checkouts.sf.yml (hunk 3 above)
Comment (valentindellangela) [reply reference/checkouts.sf.yml:1105]: Thanks - I wonder if we should just use the same as V3 `Display name of the coupon.` 🤷‍♂️

**Review by valentindellangela — APPROVED** [review]

**Review by bc-traciporter — APPROVED** [review]

**Pull request merged**